spring-batch-integration/src/main/java/org/springframework/batch/integration/config/annotation/BatchIntegrationConfiguration.java

```diff
@@ -56,4 +63,16 @@ public RemoteChunkingMasterStepBuilderFactory remoteChunkingMasterStepBuilderFac
 		return new RemoteChunkingWorkerBuilder<>();
 	}
 
+	@Bean
+	public RemotePartitioningMasterStepBuilderFactory remotePartitioningMasterStepBuilderFactory() {
+		return new RemotePartitioningMasterStepBuilderFactory(this.jobRepository,
+				this.jobExplorer, this.transactionManager);
+	}
+
+	@Bean
+	public RemotePartitioningWorkerStepBuilderFactory remotePartitioningWorkerStepBuilderFactory() {
```

Comment:
Didn't you think about separating concerns for high-level representation?
I mean I might be interested only in `master` or only in `worker` in my current microservice. Therefore extra beans are just noise for me. 

Am I missing anything ?

Reply:
If you are interested only in the master or worker, then you can inject only beans related to the master or worker setup. Some people prefer declaring all beans in the same config class and use profiles to activate/deactivate some beans.

Reply:
Right, I can inject, but my application context is still polluted with all of them.
The problem that you don't give me any preferences. There is only one annotation to add on my `@Configuration` and both infrastructure are going to be available in my application context.

Not sure what you mean about profiles, but this is already out of my control since we are talking here about imported configuration from the annotation without any profiling.

That might not be so bad since this factories are not big objects at all. I might be a bit confused eventually when I have to provide some infrastructure bean which is required by the `master`, when I am only in the `worker` or vise versa.

Just my feeling with the hope you get some food to think in the future.
The current solution might be just OK as is. Just be prepared for some similar feedback from the community! 😄

Reply:
I see your point. What do you suggest to separate the concerns? Two annotations? An additional attribute in the current annotation? Any other idea is welcome!

For the record, `@EnableBatchProcessing` adds 9 beans to the application context and I might not be interested in all of them. We took the same approach and we thought it's probably a good idea to centralize the 4 beans related to integration under the `@EnableBatchIntegration` config.

Reply:
That's fine with me. I just want to warn you about potential requests from the community.
Or even better - document the situation since we are already aware of the concern.
That's all, nothing more. Even `@EnableIntegration` provides a lot of beans. My only concern here that there is a clear separation of `master` and `worker` responsibility, so that might be a bit confusing here, unlike in the Integration when all the stuff is enabled by default, but that doesn't say that it is only for `client` or `server` or similar separation of responsibility.

Let's stop from here and go on with your current solution!